TestGeometricSeqPhenotype.java

```diff
@@ -0,0 +1,199 @@
+        assertEquals(Parameters.startingSequence.length(), emptyPheno.getSequence().length());
+
+        assertEquals("TGCATC", simplePheno.getSequence());
+    }
+
+    /**
+     * Test getter method.
+     */
+    @Test
+    public void testGetTraits() {
+        assertEquals(0.0, simplePheno.getTraitA(), 0);
+        assertEquals(0.0, simplePheno.getTraitB(), 0);
+    }
+
+    /**
+     * Test getter method.
+     */
+    @Test
+    public void testGetSequence() {
+        assertEquals("TGCATC", simplePheno.getSequence());
+    }
+
+    @Test
+    public void testDistance() {
+        assertEquals(2.236068, simplePheno.distance(history[0]), 0.000001);
+        assertEquals(3.605551, simplePheno.distance(history[1]), 0.000001);
+        assertEquals(9.433981, simplePheno.distance(history[2]), 0.000001);
+
+        assertEquals(7.211103, history[2].distance(history[0]), 0.000001);
+        assertEquals(7.211103, history[0].distance(history[2]), 0.000001);
+    }
+
+    /**
+     * Test mutate function.
+     */
+    @Test
+    public void testMutate() {
+        GeometricSeqPhenotype mutantPheno = (GeometricSeqPhenotype) simplePheno.mutate();
+
+        assertEquals("TGCATC", simplePheno.getSequence());
+        assertNotEquals("TGCATC", mutantPheno.getSequence());
+
+        String[] wildTypeMutantAminoAcids = simplePheno.mutateHelper(0, 'A');
+        assertEquals(new String[]{"C", "S"}, wildTypeMutantAminoAcids);
+    }
+
+    /**
+     * Test riskOfInfection calculations.
+     */
+    @Test
+    public void testRiskOfInfection() {
+        // closestDistance * Parameters.smithConversion = 2.236068 * 0.1 = 0.2236068
+        // minRisk = 1.0 - Parameters.homologousImmunity = 1.0 - 0.95 = 0.05
+        // risk = Math.max(minRisk, risk) = max(0.05, 0.2236068) = 0.2236068
+        // risk = Math.min(1.0, risk) = 0.2236068
+        assertEquals(0.2236068, simplePheno.riskOfInfection(history), 0.00001);
+
+        // closestDistance * Parameters.smithConversion = 15.132746 * 0.1 = 1.5132746
+        // minRisk = 1.0 - Parameters.homologousImmunity = 1.0 - 0.95 = 0.05
+        // risk = Math.max(minRisk, risk) = max(0.05, 1.5132746) = 1.5132746
+        // risk = Math.min(1.0, risk) = 1.0
+        assertEquals(1.0, simplePheno2.riskOfInfection(history), 0.0);
+    }
+
+    /**
+     * A redundant test for returning the sequence, but with toString().
+     */
+    @Test
+    public void testToString() {
+        // This may change as the class develops (i.e., new enhancements).
+        assertEquals("TGCATC, 0.0000, 0.0000, 0, 0", simplePheno.toString());
+        assertNotEquals("TGCATC, 0.0000, 0.0000, 0, 0", simplePheno.mutate().toString());
+    }
+
+    /**
+     * Test that DMS data stored in Antigen reflects the data passed in.
+     */
+    @Test
+    public void testDMSData() throws FileNotFoundException {
+        int numberOfAminoAcidSites = Parameters.startingSequence.length() / 3;
```

Comment:
Could you please add a few comments that describe what you're testing in a bit more detail?